File: twoSum.go
package main

import (
	"fmt"
)

func main() {
	result := twoSum([]int{2, 7, 3, 0}, 2)

	fmt.Print(result)
}

func twoSum(nums []int, target int) [2]int {
	m := make(map[int]int)

	for i, v := range nums {
		complement := target - v
		containIndex, contains := m[complement]

		if contains {
			return [2]int{containIndex, i}
		}

		m[v] = i
	}

	return [2]int{0, 0}
}

File: src/main/java/tj/mirboboev/challenges/TwoSum.java
package tj.mirboboev.challenges;

import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seenNumbers = new HashMap<>();
        int[] result = new int[2];

        for (int i = 0; i < nums.length; i++) {
            if (seenNumbers.containsKey(target - nums[i])) {
                return new int[]{i, seenNumbers.get(target - nums[i])};
            } else {
                seenNumbers.put(nums[i], i);
            }
        }

        return result;
    }
}
